api/src/test/java/org/openmrs/module/ModuleUtilTest.java

```diff
@@ -730,4 +736,39 @@ protected File getEmptyJarDestinationFolder() throws IOException {
 		}
 		return destinationFolder;
 	}
+	
+	@Test
+	public void file2url_shouldReturnNullIfFileIsNull() throws MalformedURLException {
+
+		Assert.assertNull(ModuleUtil.file2url(null));
+	}
+	
+	@Test
+	public void getPackagesFromFile_shouldReturnEmptyStringSetIfNonJarFile() {
+		
+		File f = new File(this.getClass()
+			.getResource("/org/openmrs/module/include/test1-1.0-SNAPSHOT.omod").getFile());
+		Assert.assertTrue(ModuleUtil.getPackagesFromFile(f).isEmpty());
+	}
+	
+	@Test
+	public void getPackagesFromFile_shouldReturnPackagesWithinJar() throws IOException{
+
+		String prefix = "org.openmrs.module.test1";
+		String[] expectedPackages = new String[]{ prefix+".api.impl", prefix+".api", prefix, prefix+".api.db.hibernate",
+			prefix+".api.db", prefix+".extension.html", prefix+".web.controller" };
+		
+		File srcFile = new File(this.getClass()
+			.getResource("/org/openmrs/module/include/test1-1.0-SNAPSHOT.omod").getFile());
+		File destFile = new File("/tmp/test1-1.0-SNAPSHOT.jar");
+		FileUtils.copyFile(srcFile, destFile);
+				
+		HashSet<String> packages = (HashSet<String>)ModuleUtil.getPackagesFromFile(destFile);
+		
+		Files.delete(Paths.get(destFile.getAbsolutePath()));
```

Comment:
I suggest you use junit rule https://github.com/junit-team/junit4/wiki/Rules#temporaryfolder-rule

which takes care of creating/cleaning up temp folders and files

see how its used
https://github.com/openmrs/openmrs-core/blob/master/api/src/test/java/org/openmrs/module/ModuleFileParserTest.java#L69-L70